code/modules/mob/living/living.dm

```diff
@@ -1495,3 +1495,35 @@
 	if(!apply_change)
 		return BODYTEMP_NORMAL
 	return BODYTEMP_NORMAL + get_body_temp_normal_change()
+
+/mob/living/proc/can_look_up() //Checks if the user is incapacitated.
+	return !((next_move > world.time) || incapacitated(ignore_restraints = TRUE))
+
+/**
+ * look_up Changes the perspective of the mob to any openspace turf above the mob
+ *
+ * This also checks if an openspace turf is above the mob before looking up or resets the perspective if already looking up
+ *
+ */
+/mob/living/verb/look_up()
+	set name = "Look Up"
+	set category = "IC"
+
+	if(client.perspective != MOB_PERSPECTIVE) //We are already looking up.
+		reset_perspective()
+		UnregisterSignal(src, COMSIG_MOVABLE_PRE_MOVE)
+		return
+	if(!can_look_up())
+		return
+	var/turf/ceiling= get_step_multiz(src, UP)
+	if(!ceiling) //We are at the highest z-level.
+		to_chat(src, "<span class='warning'>You can't see through the ceiling above you.</span>")
+		return
+	else if(!isopenspace(ceiling)) //There is no openspace turf above us.
+		to_chat(src, "<span class='warning'>You can't see through the floor above you.</span>")
```

Comment:
If you really wanted to you could add a check to allow viewing past the floor if you had something like thermals/xray vision

Reply:
no he can't because that doesn't work with how this works.